### PR TITLE
Added first user facing contacts page

### PR DESCRIPTION
**Pull request by raeeschachar**

### CODE REVIEW

**Comment by raeeschachar** [comment]
@MaryKhan 
Please review

File: contacts/views.py [hunk 1]
```diff
@@ -1,5 +1,33 @@
...
+#     template_name = 'contacts/detail.html'
+#
+#     def get_queryset(self):
+#         """
+#         Return details of contact
+#         """
+#         return Contact.objects.get(id=Contact.id)
+
+
+def detail(request, contact_id):
```
Comment (MaryKhan) [inline contacts/views.py]: i would recommend to use class-based views rather than functions.

File: contacts/views.py [hunk 2]
```diff
@@ -1,5 +1,33 @@
...
+    contact_details = Contact.objects.get(id=contact_id)
+    address_details = Address.objects.get(id=contact_id)
```
Comment (MaryKhan) [inline contacts/views.py]: the query to `Address` model will fetch wrong results, think on the mistake you're doing here! :P

File: phone_directory/settings.py [hunk 3]
```diff
@@ -31,7 +31,7 @@
 # Application definition
 
 INSTALLED_APPS = [
-    'contacts',
+    'contacts.apps.MyDirectoryConfig',
```
Comment (MaryKhan) [inline phone_directory/settings.py:34]: why did we need this change here?

File: contacts/templates/contacts/contact.html [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>My Contacts</title>
+</head>
+<body>
+<h1>My Contacts</h1>
+
+    {% for name in my_contacts %}
```
Comment (MaryKhan) [inline contacts/templates/contacts/contact.html]: the items in `my_contacts`list are not just `name` but `contact`, change the var here!

File: contacts/templates/contacts/contact.html [hunk 5]
```diff
@@ -0,0 +1,18 @@
+{% load staticfiles %}
+
+<link rel="stylesheet" type="text/css" href="{% static 'contacts/style.css' %}" />
```
Comment (MaryKhan) [inline contacts/templates/contacts/contact.html]: css/js imports should be in header of the html page.

File: contacts/templates/contacts/detail.html [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+<body>
+<h1>{{ contact_details }}</h1>
+<p><b>Email: </b>{{ contact_details.email }}</p>
+<p><b>Phone: </b>{{ contact_details.phone_number }}</p>
+
+<p><b>Address: </b>
+    {{ address_details.address_line }}, {{ address_details.city }},
+    {{ address_details.state }}, {{ address_details.zip_code }}</p>
+<p><b>Country: </b>{{ address_details.country }}</p>
+</body>
```
Comment (MaryKhan) [inline contacts/templates/contacts/detail.html]: pep8: extra white space needed.

File: contacts/urls.py [hunk 7]
```diff
@@ -2,6 +2,10 @@
 
 from . import views
 
+
+app_name = 'contacts'
 urlpatterns = [
-    url(r'^$', views.index, name='index')
+    url(r'^$', views.ContactView.as_view(), name='contact'),
+    # url(r'^(?P<pk>[0-9]+)/$', views.DetailView.as_view(), name='detail'),
+    url(r'^([0-9]+)/$', views.detail, name='detail')
```
Comment (MaryKhan) [inline contacts/urls.py]: the ending slash should be optional in url.

File: contacts/views.py [hunk 8]
```diff
@@ -1,5 +1,33 @@
-from django.http import HttpResponse
+from django.shortcuts import render
+from django.views import generic
+from .models import Contact, Address
 
 
-def index(request):
-    return HttpResponse("Sample View")
+class ContactView(generic.ListView):
```
Comment (MaryKhan) [inline contacts/views.py]: good one! 👍

**Review by MaryKhan — CHANGES_REQUESTED** [review]

File: phone_directory/settings.py (hunk 3 above)
Comment (raeeschachar) [reply phone_directory/settings.py:34]: To use latest django and a better way as per django docs.

File: contacts/urls.py (hunk 7 above)
Comment (raeeschachar) [reply contacts/urls.py]: done

File: contacts/templates/contacts/detail.html (hunk 6 above)
Comment (raeeschachar) [reply contacts/templates/contacts/detail.html]: done

File: contacts/templates/contacts/contact.html (hunk 4 above)
Comment (raeeschachar) [reply contacts/templates/contacts/contact.html]: done

File: contacts/templates/contacts/contact.html (hunk 5 above)
Comment (raeeschachar) [reply contacts/templates/contacts/contact.html]: done

File: contacts/views.py (hunk 2 above)
Comment (raeeschachar) [reply contacts/views.py]: done

File: contacts/urls.py [hunk 9]
```diff
@@ -2,6 +2,10 @@
 
 from . import views
 
+
+app_name = 'contacts'
 urlpatterns = [
-    url(r'^$', views.index, name='index')
+    url(r'^$', views.ContactView.as_view(), name='contact'),
+    # url(r'^(?P<pk>[0-9]+)/$', views.DetailView.as_view(), name='detail'),
+    url(r'^([0-9]+)$', views.detail, name='detail')
```
Comment (MaryKhan) [inline contacts/urls.py]: wrong attempt! think again!

File: contacts/templates/contacts/detail.html [hunk 10]
```diff
@@ -0,0 +1,19 @@
+{% load staticfiles %}
+
+<link rel="stylesheet" type="text/css" href="{% static 'contacts/style.css' %}"/>
```
Comment (MaryKhan) [inline contacts/templates/contacts/detail.html]: same deal!
also, can we have a layout html to put common markups at one place?

**Review by MaryKhan — CHANGES_REQUESTED** [review]

**Comment by raeeschachar** [comment]
@MaryKhan good to merge?

**Comment by MaryKhan** [comment]
yep, but do study and understand why we did this:
https://github.com/raeeschachar/phone_directory/pull/3/files#diff-f34afa639824b61e360a1cb7f245a05eR8
https://github.com/raeeschachar/phone_directory/pull/3/files#diff-747ebed9547a42c9e23ecda8101269a8R9

**Pull request merged**